server/autolinkapp/form.go

```diff
@@ -0,0 +1,277 @@
+	call := &apps.CallRequest{}
+	err := json.NewDecoder(r.Body).Decode(call)
+	if err != nil {
+		httputils.WriteJSON(w, apps.NewErrorCallResponse(errors.Wrap(err, "failed to decode call request body")))
+		return
+	}
+
+	values, err := extractFormValues(call.Values)
+	if err != nil {
+		httputils.WriteJSON(w, apps.NewErrorCallResponse(errors.Wrap(err, "error extracting form values")))
+		return
+	}
+
+	if call.SelectedField == fieldLink {
+		if values.Link.Value == createOptionValue || values.Link.Value == "" {
+			httputils.WriteJSON(w, a.getEmptyFormResponse())
+			return
+		}
+
+		links := a.store.GetLinks()
+		for _, link := range links {
+			if link.Name == values.Link.Value {
+				values = formValuesFromAutolink(link)
+				break
+			}
+		}
+	}
+
+	f := a.getForm(values)
+	resp := apps.CallResponse{
+		Type: apps.CallResponseTypeForm,
+		Form: f,
+	}
+
+	httputils.WriteJSON(w, resp)
+}
+
+func (a *app) getForm(values FormValues) *apps.Form {
+	createNewOption := apps.SelectOption{
+		Label: "(create new)",
+		Value: createOptionValue,
+	}
+
+	links := a.store.GetLinks()
+	sort.Slice(links, func(i, j int) bool {
+		return strings.Compare(strings.ToLower(links[i].Name), strings.ToLower(links[j].Name)) < 0
+	})
+
+	linkOptions := []apps.SelectOption{createNewOption}
+	for _, link := range links {
+		linkOptions = append(linkOptions, apps.SelectOption{
+			Label: link.Name,
+			Value: link.Name,
+		})
+	}
+
+	linkOption := createNewOption
+	if values.Link.Value != "" {
+		linkOption = values.Link
+	}
+
+	return &apps.Form{
+		Title:         "Edit Autolinks",
+		SubmitButtons: fieldSubmitButtons,
+		Call:          apps.NewCall("/edit-links"),
+		Fields: []*apps.Field{
+			{
+				Name:                fieldLink,
+				ModalLabel:          "Link",
+				Type:                apps.FieldTypeStaticSelect,
+				Value:               linkOption,
+				SelectStaticOptions: linkOptions,
+				SelectRefresh:       true,
+			},
+			{
+				Name:       fieldName,
+				ModalLabel: "Name",
+				Type:       apps.FieldTypeText,
+				Value:      values.Name,
+				IsRequired: true,
+			},
+			{
+				Name:       fieldEnabled,
+				ModalLabel: "Enabled",
+				Type:       apps.FieldTypeBool,
+				Value:      values.Enabled,
+			},
+			{
+				Name:        fieldWordMatch,
+				ModalLabel:  "Word Match",
+				Type:        apps.FieldTypeBool,
+				Value:       values.WordMatch,
+				Description: "If true uses the [word boundaries](https://www.regular-expressions.info/wordboundaries.html)",
+			},
+			{
+				Name:       fieldDisableNonWordPrefix,
+				ModalLabel: "Disable Non-word Prefix",
+				Type:       apps.FieldTypeBool,
+				Value:      values.DisableNonWordPrefix,
+			},
+			{
+				Name:       fieldDisableNonWordSuffix,
+				ModalLabel: "Disable Non-word Suffix",
+				Type:       apps.FieldTypeBool,
+				Value:      values.DisableNonWordSuffix,
+			},
+			{
+				Name:       fieldScope,
+				ModalLabel: "Scope",
```

Comment:
Out of scope of this PR: This should be autocompleted

Reply:
I'm not sure how that would be autocompleted. Can you explain further?

Reply:
`Scope` is either a channel or a team.